### ISSUE TITLE
Code block fencing fails on enTangleD own configuration

### ISSUE THREAD

**Issue by mgajda** [entangled/entangled#74]
Past detection of code block fences failed on the blocks that are indented, since it assumed that we only have one started at the start of line like this:
```
   1. You may use this code by default:
        ```{.haskell #default-variant}
        main = hspec $ do 
        ```
   2. To set custom settings use this instead:
       ```{.haskell #custom-settings-variant}
       main hspecWith customSettings $ do
       ```
```

To fix this, we currently use regex that fails on `entangled.dhall` as defined in `lit/04-configuration.md`, and as a result enTangleD fails to build itself.
``` {.dhall file=data/config-schema.dhall}
    { matchCodeStart       = "^[ ]*```[ ]*{[^{}]*}"
    , matchCodeEnd         = "^[ ]*```"
    , extractLanguage      = "^[ ]*```[ ]*{\\.([^{} \t]+)[^{}]*}"
    , extractReferenceName = "^[ ]*```[ ]*{[^{}]*#([^{} \t]*)[^{}]*}"
    , extractFileName      = "^[ ]*```[ ]*{[^{}]*file=([^{} \t]*)[^{}]*}" }
```
To fix this we should only allow spacing after beginning of the line `^` and before ` ``` `.

**Comment by justjosias** [comment]
See also: #60.

**Comment by mgajda** [comment]
Fixed in #75.
